### ISSUE TITLE
full transparent Vortex Mod Manager support

### ISSUE THREAD

**Issue by frostworx** [sonic2kk/steamtinkerlaunch#25]
I'm working on implementing full transparent Vortex Mod Manager support "for steam on linux"
https://github.com/Nexus-Mods/Vortex/issues/2862
The ultimate goal is that the user only has to add any (Vortex supported) game into a steam "Vortex" category and everything else is setup completely automatically:
initial global installation of Vortex and dependencies, create registry entries, paths, symlinks, preconfigure global and gamespecific settings in Vortex, add nxm mime-types and so on. So the prelaunched Vortex will be (hopefully) ready to use for the currently started game. It won't be possible to start the game itself via Vortex (different WINEPREFIX), but the regular game start will start after exiting Vortex anyway, so this shouldn't be an issue.
It still has many rough edges, and therefore is not in repo yet, but basically it works pretty good already.
The only show-stopper  problem is that enabled mods are not created in the games mod dir, which makes the whole construct pretty pointless atm.
There are several nexus deployment methods https://wiki.nexusmods.com/index.php/Deployment_Methods
and looks like softlinks would be the best choice in theory for linux.
This is very likely only an issue with games on a different partition, but there must be some way to workaround this problem.

Haven't tested if bind mounting would workaround this, but it can't be a soluton to depend on root priviledges.
Also parsing the Vortex created "vortex.deployment.json" to create the symlinks after exiting Vortex and before launching the game can't be the best/a valid solution.

If someone has an idea feel free to reply here

**Comment by frostworx** [comment]
I want Vortex integration as transparent and zeroconf as possible.

General usage of Vortex is enabled with the VORTEX variable set to 1 (f.e. enabled per game via "Vortex" steam category)
If set to 1, what should be the default options:

- variable VORTEXPREFIX is used as WINEPREFIX for Vortex - should this be preconfigured per default (and if so where?) or undefined?
- should the vortex-setup exe automatically be downloaded or accepted with another variable (f.e. VORTEXDOWNLOAD)?
- I consider leaving the variable VORTEXWINE undefined per default - if it is not defined by the user, system "wine" is used when available and if not fall back to currently running proton-wine (not sure yet if this works without any problems)
- if there's no sane way to workaround the symlink problem above, maybe syncing/using between multiple VORTEXPREFIX on all configured SteamLibrary partitions before starting Vortex would be a valid (but ugly) solution

What do you think?

**Comment by frostworx** [comment]
yay, simply using the hardlink_activator fixes the "symlink problem" - should have tried that at the first place.
back to finetuning Vortex support.
leaving this issue open for the other points above

**Comment by frostworx** [comment]
Hi @frostwork
I have another cool idea - variable VORTEX should have 4 different modes:

- VORTEX=0: Vortex is completely ignored - game starts normally
- VORTEX=1: Vortex will be installed from scratch if not found, stl will configure links,registry,settings,dirs etc required for Vortex, start Vortex and will start the game after exiting Vortex (default mode when started via Vortex steam category)
- VORTEX=2: Vortex will start without any checks (assuming it is completely ready) and will start the game after exiting Vortex
- VORTEX=3: Vortex will start without any checks, but game won't be started after exiting Vortex

**Comment by frostworx** [comment]
Great idea, @frostworx 
Just implemented it

**Comment by frostworx** [comment]
@ https://github.com/frostworx/steamtinkerlaunch/issues/25#issuecomment-672199611
no, this doesn't work completely as well:
https://github.com/Nexus-Mods/Vortex/issues/2862#issuecomment-672122884

**Comment by frostworx** [comment]
found a valid workaround:
```
GAMEMP="$(df -P "GP" | awk 'END{print $NF}')"
writelog "$(date) - INFO - ${FUNCNAME[0]} - mount point of partition where the game is installed: $GAMEMP"
# TODO: autofind a dir from array VORTEXSTAGES on the same $GAMEMP partition like the game is installed and use that as VORTEXSTAGING
```

**Comment by frostworx** [comment]
Vortex implementation already works excellent
Tested almost 30 games and all work ootb seamlessly without any user configuration required (but optionally possible of course).
Some more finetuning and cleanups before release.
It is a shame though that Script Extender for Skyrim, Fallout flavours apparantly no longer works with proton 5.0.
I already implemented an automated start of script extender instead of the game exe, but it would help if the exe would work in proton...
https://github.com/ValveSoftware/Proton/issues/170
